addons/base_vat/models/res_partner.py

```diff
@@ -93,42 +95,41 @@ def vies_vat_check(self, cr, uid, country_code, vat_number, context=None):
             # TIMEOUT or SERVER_BUSY. There is no way we can validate the input
             # with VIES if any of these arise, including the first one (it means invalid
             # country code or empty VAT number), so we fall back to the simple check.
-            return self.simple_vat_check(cr, uid, country_code, vat_number, context=context)
+            return self.simple_vat_check(country_code, vat_number)
 
-    def check_vat(self, cr, uid, ids, context=None):
-        user_company = self.pool.get('res.users').browse(cr, uid, uid).company_id
+    @api.multi
+    @api.constrains("vat")
+    def check_vat(self):
+        user_company = self.env.user.company_id
         if user_company.vat_check_vies:
             # force full VIES online check
             check_func = self.vies_vat_check
         else:
             # quick and partial off-line checksum validation
             check_func = self.simple_vat_check
-        for partner in self.browse(cr, uid, ids, context=context):
+        for partner in self:
             if not partner.vat:
                 continue
             vat_country, vat_number = self._split_vat(partner.vat)
-            if not check_func(cr, uid, vat_country, vat_number, context=context):
+            if not check_func(vat_country, vat_number):
                 _logger.info(_("Importing VAT Number [%s] is not valid !" % vat_number))
-                return False
+                partner._construct_constraint_msg()
         return True
 
-    def _construct_constraint_msg(self, cr, uid, ids, context=None):
+    @api.multi
+    def _construct_constraint_msg(self):
         def default_vat_check(cn, vn):
             # by default, a VAT number is valid if:
             #  it starts with 2 letters
             #  has more than 3 characters
             return cn[0] in string.ascii_lowercase and cn[1] in string.ascii_lowercase
-        vat_country, vat_number = self._split_vat(self.browse(cr, uid, ids)[0].vat)
+        vat_country, vat_number = self._split_vat(self.vat)
```

Comment:
Use self.ensure_one() to verify that the self contains only one record or make the method to work on multiple records.